投稿機能の非同期機能(chat space)

doctaignorantia opened the pull request:
#実装内容について  
##1.Controllerからhtmlをレスポンスとするコードを削除しても問題なく動作するか（非同期通信がそもそも出来ているのか）  
 非同期通信はできていると思います。htmlをレスポンスとするコードを削除については、不要と思われたところを削除したところjqery.jsと連動していたので削除していません。    
##2.画面がリロードせずに投稿できているか  
 はい。投稿できてます。  
##3.テキストのみ/画像のみ/画像とテキスト、それぞれでエラーが出ずに投稿ができるか  
連続での投稿が可能か  
はい。連続投稿可能です。  
##4.投稿と同時に最下部までスクロールするか  
はい。最下部までスクロールします。  
##5.日時表記は適切か  
はい。日本の時間帯で表記されます。  
##6.関数名は処理の内容を過不足なく表現できるような名前になっているか  
はい。分かるように表現できてると思います。  
##7.変数名はデータの内容を過不足なく表現できるような名前になっているか  
はい。分かるように表現できてると思います。  
##8.SENDボタンを連打したら適切なアラートが表示されるか  
はい。アラートが表示されます。  
##9.コンソール上にエラーが出ていないか  
はい。エラーはありません。  
##10.binding.pryやconsole.logなどのデバッグ用の記載は消去してあるか  
はい。消去致しました。  
##11.コードのインデントは適切に取ってあるか（全角スペースはNGです）  
はい。全角スペースはありません。インデントも問題ないと思われます。(hamlだけなぜかインデントが4に設定されていたので4になっています。)  
[画面収録 2019-09-06 17.12.49.mov.zip](https://github.com/doctaignorantia/chat-space/files/3582950/2019-09-06.17.12.49.mov.zip)

#プルリクエストの作成について
##1.プルリクエストが正しく作られているか
 はい。正しく作ったと思います。
##2.プルリクエストにWhy/Whatを記載してあるか
 はい。記載が必要な個所は記載致しました。
##3.プルリクエストのWhy/Whatにマークダウンが適用されているか
 はい。一応、できていると思います。
##4.実際の動きを示すgifの動画キャプチャを添付できているか
 はい。すみませんが、movで添付致しました。

doctaignorantia committed:
first commit

doctaignorantia committed:
console.log(this);(chat-space)

とりあえず、console.log(this);で
catch

doctaignorantia committed:
未完成(chat-spacd)

非同期通信まだ

doctaignorantia committed:
非同期通信(chat-space)

とりあえずできる。
空白の時のエラーが出ない

doctaignorantia committed:
非同期

doctaignorantia committed:
indent

doctaignorantia committed:
非同期はできた(chat-space)

i-date commented on app/assets/javascripts/message.js:
```diff
@@ -0,0 +1,56 @@
+      var content = message.content ? `${ message.content }` : "";
+      var image = message.image ? `<img src= ${ message.image }>` : "";
+      var html = `
+      
```
8行目は不要な空行でしたら削除しましょう。

i-date commented on app/assets/javascripts/message.js:
```diff
@@ -0,0 +1,56 @@
+      .done(function(data){ 
+        var html = buildHTML(data);
+        $('.messages').append(html);
+        $('form__message').val('');
```
45行目でresetを使用しているため、44行目は不要ですので削除しましょう。

i-date commented on app/assets/javascripts/message.js:
```diff
@@ -0,0 +1,56 @@
+      .fail(function(){
+        alert('メッセージを入力してください。');
+      })
+      .always(() => {
```
51行目は`.always(function () {`と記述できますので、`.done〜`と`.fail〜`とフォーマットを統一しましょう。

i-date reviewed (commented):
ご提出ありがとうございます！コメントいたしました^^
また、[Files changed](https://github.com/doctaignorantia/chat-space/pull/6/files#diff-dd794c97049ca1676de61c69cf59ff6b)でご確認いただけますが、vendor/bundle以下の差分ファイルが大量にpushされてしまっています。gitignoreに追加して、github上から削除しましょう。

doctaignorantia committed:
非同期通信(chat-space)

gitignore(vendor/bundle/*)
追加
空行の削除
$('form')[0].reset();と
$('form__message').val('');との重複修正
formatの統一

doctaignorantia commented:
gitignore に
vendor/bundle/*  を追加致しました。 
 
app/assets/javascripts/message.js
8行目の空行の削除を致しました。

app/assets/javascripts/message.js 
$('form')[0].reset();と
$('form__message').val('');との機能の重複修正について
$('form__message').val('');を削除し重複を解消致しました。

app/assets/javascripts/message.js 
.always(() => { について、  
.always(function(){ としformatを
他と同様になるようの統一致しました。

i-date reviewed (approved):
修正ありがとうございます！LGTMとさせていただきます^^

doctaignorantia commented:
ありがとうございます。
